picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -689,6 +688,16 @@ public static Picasso with(@NonNull Context context) {
   }
 
   /**
+   * The global default {@link Picasso} instance.
+   * <p>
+   * This instance is created by passing a global context from startup into {@link #with(Context)}.
+   * </p>
+   */
+  public static Picasso with() {
+    return with(PicassoProvider.context);
```

Comment:
nice, still lazy and still lets setSingletonInstance work fine without disabling anything.